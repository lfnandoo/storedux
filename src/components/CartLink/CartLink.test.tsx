import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

import { CartLink } from "./index";

describe("CartLink", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <CartLink productsInCart={1} />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should get cart button", () => {
    const { getByTestId } = render(<CartLink productsInCart={1} />, {
      wrapper: BrowserRouter,
    });

    expect(getByTestId("cart-btn")).toBeInTheDocument();
  });

  it("should not get cart button", () => {
    const { container } = render(<CartLink productsInCart={0} />, {
      wrapper: BrowserRouter,
    });

    expect(container).toBeEmptyDOMElement();
  });

  it("should redirect to cart page", () => {
    const cartPath = "/cart";
    const { getByTestId } = render(<CartLink productsInCart={1} />, {
      wrapper: BrowserRouter,
    });

    fireEvent.click(getByTestId("cart-btn"));

    const url = document.URL;
    const uri = url.length - cartPath.length;
    const path = url.slice(uri);

    expect(path).toBe(cartPath);
  });
});
