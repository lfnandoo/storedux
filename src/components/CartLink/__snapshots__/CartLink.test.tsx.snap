// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartLink should render correctly 1`] = `
<a
  className="sc-bdnxRM cjgSfr"
  data-testid="cart-btn"
  href="/cart"
  onClick={[Function]}
>
  <img
    alt="cart"
    src="shopping-cart-outline.svg"
  />
</a>
`;
